### DISCUSSION TITLE
podget use with mpd/mpd

### DISCUSSION THREAD

**Discussion by theorbitofmercury** [dvehrs/podget#57] [Show and tell]
Found podget to be the easiest/simplest pod catcher for my Raspberry Pi(s)/mpd/snapcast constellation. I had to do some work on the New playlists. So:
1) mounted the default POD directory in the mpd music directory. Then ncmpcpp(etc) mpd clients can then pick up the downloaded pods after a mpc update of its library (podget and mpc update in crontab).
2) the new playlists don't have the full path of the pods. So had to use sed to prefix each line with the missing path. Had to do a grep though to get only the m3u's which had not already got the full path added(otherwise the path keeps getting added as a prefix to each line again)
3) also the new playlists have a double // which mpd doesn't like. So another sed.

crontab:
```
15 04 * * * /usr/bin/podget -r 1 --silent
30 04 * * * sed -i -e 's/\/\//\//g' /home/pi/POD/*.m3u
31 04 * * * grep -rL /home/ /home/pi/POD/*.m3u | xargs sed -i -e 's#^#/home/pi/Music/mpd_mnt/RPi4-pod/#'
32 04 * * * mpc update
``` 
Hope this of use to someone else.

**Comment by dvehrs** [comment]
This is an interesting suggestion but there might be one small issue.   What if the podget session takes more than 15 minutes?  That delay could affect what needs to happen to the files.  Might be better to put all this into one wrapper script then set it as a cronjob.

FILE: /usr/local/bin/mypodget.sh
```
#!/usr/bin/env bash
/usr/bin/podget -r 1 --silent
sed -i -e 's/\/\//\//g' /home/pi/POD/*.m3u
grep -rL /home/ /home/pi/POD/*.m3u | xargs sed -i -e 's#^#/home/pi/Music/mpd_mnt/RPi4-pod/#'
mpc update
```

crontab:
```
15 04 * * * /usr/local/bin/mypodget.sh
```

Now this should work but may interfere with any automatic cleanup that Podget attempts to do because of the path changes in the playlist.

**Reply by theorbitofmercury** [reply]
The 15min gap is arbitrary. I could make it 60 min to make sure the podget downloads are always finished. I don't understand the relationship between playlists and automatic cleanup. Creating playlists is optional? So does the cleanup depend on the playlist being created? That's not clear in the documentation. http://manpages.ubuntu.com/manpages/bionic/man7/podget.7.html.

Cheers Mike

**Reply by dvehrs** [reply]
OK, now you're mixing two threads that no one else will be able to follow.  

As I already conceded in the other thread ( https://github.com/dvehrs/podget/issues/58 ), the documentation is less than ideal on this facet of podget use but to repeat myself here for anyone viewing this discussion:

Automatic cleanup uses the playlists to determine which files to remove.  It does this because it is somewhat dangerous to use file modification (or change) dates because when we download from a few types of servers those dates are maintained from the original.  This is good because it makes it easier to keep files in the order they were created but bad because it can make it more difficult for someone who has playlists disabled to clean out older files.  

Dave

**Comment by theorbitofmercury** [comment]
see the issue thread  ( #58 ) for further stuff
